presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -370,14 +371,20 @@ protected Node visitShowCreate(ShowCreate node, Void context)
             QualifiedObjectName objectName = createQualifiedObjectName(session, node, node.getName());
             Optional<ViewDefinition> viewDefinition = metadata.getView(session, objectName);
 
+            checkObjectAccessible(node, objectName);
+
             if (node.getType() == VIEW) {
                 if (!viewDefinition.isPresent()) {
-                    if (metadata.getTableHandle(session, objectName).isPresent()) {
+                    if (isTableAccessible(objectName)) {
                         throw new SemanticException(NOT_SUPPORTED, node, "Relation '%s' is a table, not a view", objectName);
                     }
                     throw new SemanticException(MISSING_TABLE, node, "View '%s' does not exist", objectName);
                 }
 
+                if (!isViewAccessible(objectName)) {
+                    throw new SemanticException(NOT_SUPPORTED, node, "View '%s' does not exist", objectName);
```

Comment:
new code path? new test?

Reply:
This check is redundant. I am going to remove it.